actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/service/UserProfileReadService.java

```diff
@@ -415,6 +423,41 @@ public void addExtraFieldsInUserProfileResponse(
     }
   }
 
+  private void addSchoolLocation(Map<String, Object> result, RequestContext context) {
+    String rootOrgId = (String) result.get(JsonKey.ROOT_ORG_ID);
+    List<Map<String, Object>> organisations =
+        (List<Map<String, Object>>) result.get(JsonKey.ORGANISATIONS);
+    if (CollectionUtils.isNotEmpty(organisations) && organisations.size() > 1) {
+      for (int i = 0; i < organisations.size(); i++) {
+        String organisationId = (String) organisations.get(i).get(JsonKey.ORGANISATION_ID);
+        if (StringUtils.isNotBlank(organisationId) && !organisationId.equalsIgnoreCase(rootOrgId)) {
+          Map<String, Object> searchQueryMap = new HashMap<>();
+          searchQueryMap.put(JsonKey.NAME, organisations.get(i).get(JsonKey.ORG_NAME));
```

Comment:
@amit-tarento : i think we can even add type: school in the searchquerymap, so searching will take less time

Reply:
Added